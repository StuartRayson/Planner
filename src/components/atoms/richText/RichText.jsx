import Parser from 'html-react-parser';

const RichText = ({ html = '' }) => Parser(html);

export default RichText;
