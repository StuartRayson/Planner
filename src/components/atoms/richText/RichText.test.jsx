import React from 'react';
import { shallow } from 'enzyme';

import RichText from './RichText.jsx';

describe('<RichText />', () => {
  it('should render without crashing', () => {
    shallow(<RichText />);
  });

  it('should render with correct markup', () => {
    const text = 'Lorem ipsum dolor';
    const richtext = shallow(<RichText html={text} />);
    expect(richtext.text()).toEqual(text);
  });

  it('should render and match snapshot with basic html input', () => {
    const html = '<div class="example"><h1>Title</h1><p>Content</p></div>';
    const wrapper = shallow(
      <RichText html={html} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
