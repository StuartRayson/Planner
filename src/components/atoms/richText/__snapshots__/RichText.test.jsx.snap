// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichText /> should render and match snapshot with basic html input 1`] = `
<div
  className="example"
>
  <h1
    key="0"
  >
    Title
  </h1>
  <p
    key="1"
  >
    Content
  </p>
</div>
`;
